### PR TITLE
Upgrade Rails version to 6.0.3

### PR DESCRIPTION
**Pull request by katafrakt**
### What does this code do, and why?

Upgrades Rails to newest available version, in order to take advantage of latest version of the framework and get rid of some deprecation warnings (as described in #507).

### How is this code tested?

Uses regular test suite of the project.

### Are any database migrations required by this change?

No.

### Screenshots (before/after)

No visual changes detected during a brief smoke test.

### Are there any configuration or environment changes needed?

No.

## Notes

There are still a few deprecation warnings left. These come from some external dependencies and, surprisingly, from Rails itself. But the total number of warnings is greatly reduced.

Note that it also uses new code loading from Rails 6. Perhaps it's too much for one jump on for the time being it should be turned off?

### CODE REVIEW

File: app/assets/config/manifest.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+//= link admin.css
+//= link dues.js
+//= link membership_note.js
+//
+//= link application.css
+//= link application.js
```
Comment (ircarreira) [inline app/assets/config/manifest.js:6]: newline is missing (surprised rubocop didn't complain about this)

**Review by ircarreira — COMMENTED** [review]
well done!

File: app/assets/config/manifest.js (hunk 1 above)
Comment (katafrakt) [reply app/assets/config/manifest.js]: I guess Rubocop does not check JS files :wink: Fixed.

**Review by anaulin — APPROVED** [review]
I did some additional spot-checks locally, and it all seems fine. Thanks for this work, @katafrakt , super excited at having a greatly reduced warning volume! 🚀  🥦

**Pull request merged**